rpc/server.go

```diff
@@ -0,0 +1,53 @@
+package rpc
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+
+	"github.com/gluster/glusterd2/context"
+	"github.com/gluster/glusterd2/peer"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/kshlm/pbrpc/pbcodec"
+)
+
+func (r *Connection) PeerAddRPCSvc(args *peer.PeerAddRequest, reply *RPCPeerAddResp) error {
```

Comment:
The error is here. You should be the protobuf generated type RPCPeerAddReq. You changed it in all other places, except this.

Because of this, ReadRequestBody in pbcodec isn't able to unmarshal the request body, and returns an error. The `net/rpc` framework silently ignores this and doesn't log or pass up any errors.